app/routes/bookmarks/$bookmarkId.edit.tsx

```diff
@@ -0,0 +1,33 @@
+import type { ActionFunction, LoaderFunction } from "@remix-run/node";
+import { redirect } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
+import { bookmarkService } from "~/features/bookmarks/bookmark.service.server";
+import type { Bookmark } from "~/features/bookmarks/bookmark.types";
+import { BookmarkForm } from "~/features/bookmarks/BookmarkForm";
+import { AppErrorBoundary } from "~/features/error/AppErrorBoundary";
+
+interface LoaderData {
+  bookmark: Bookmark;
+}
+
+export const loader: LoaderFunction = async ({ params }) => {
+  const bookmark = await bookmarkService.get(params.bookmarkId);
+
+  return {
+    bookmark,
+  } as LoaderData;
+};
+
+export default function EditBookmarkRoute() {
+  const data = useLoaderData() as LoaderData;
+  return <BookmarkForm initial={data.bookmark} />;
+}
+
+export const action: ActionFunction = async ({ request, params }) => {
```

Comment:
Because our action is reused between `new` and `edit`. Should we hoist the action to `$bookmarkId.tsx`?